src/components/common/LocationMessages.js

```diff
@@ -76,29 +58,44 @@ class LocationMessages extends Component {
     try {
       const currentLatLong = await getCurrentPosition();
       const { latitude, longitude } = currentLatLong.coords;
-      const [addressResult] = await getAddressFromLatAndLong({
-        latitude,
-        longitude
-      });
-      const { name, street, city, region, country } = addressResult;
-
-      const address = Platform.OS === 'ios' ? name : `${street} ${name}`;
-
-      const location = {
-        address,
-        city,
-        provinceName: region,
-        country,
-        latitude,
-        longitude
-      };
-
-      this.props.onUserLocationChange(location);
+
+      if (latitude && this.props.latitude !== latitude && longitude && this.props.longitude !== longitude) {
+        const [addressResult] = await getAddressFromLatAndLong({ latitude, longitude });
+        const { name, street, city, region, country } = addressResult;
+
+        const address = Platform.OS === 'ios' ? name : `${street} ${name}`;
+
+        const location = {
+          address,
+          city,
+          provinceName: region,
+          country,
+          latitude,
+          longitude
+        };
+
+        this.props.onLocationFound && this.props.onLocationFound({ updating: false, location });
+      } else if (!latitude || !longitude) {
+        this.updateUserLocation();
+      }
     } catch (error) {
       console.error(error);
     }
   };
 
+  updateUserLocation = () => {
+    if (this.props.latitude && this.props.longitude) {
+      // Esto se hace para que cuando el marker de 'UserLocation' se está mostrando en el mapa,
+      // y  después de apaga el GPS o deje de dar permisos, este marker desaparezca.
+      // PD: El borrado del mapa no es instantáneo porque no queda el compoente escuchando, pero la
+      // próxima vez que se presione el Fab y te aparezca el cartel solicitando GPS, se borra el marker
+      this.props.onLocationValueChange({ userLocation: { latitude: null, longitude: null } });
+    }
+
+    const updating = this.state.permissionStatus ? false : true;
```

Comment:
`const updating = !this.state.permissionStatus` es mas compacto y facil de leer me parece